src/main.cc

```diff
@@ -122,110 +161,95 @@ int main(int argc, char *argv[]) {
 
     int nrules, nsamples, nlabels, nsamples_chk;
     rule_t *rules, *labels;
-    rules_init(argv[0], &nrules, &nsamples, &rules, 1);
-    rules_init(argv[1], &nlabels, &nsamples_chk, &labels, 0);
+
+    if(rules_init(argv[0], &nrules, &nsamples, &rules, 1) != 0) {
+        fprintf(stderr, "Failed to load out file from path: %s\n", argv[0]);
+        return 1;
+    }
+
+    if(rules_init(argv[1], &nlabels, &nsamples_chk, &labels, 0) != 0) {
+        fprintf(stderr, "Failed to load label file from path: %s\n", argv[1]);
+        rules_free(rules, nrules, 1);
+        return 1;
+    }
+
+    if(nlabels != 2) {
+        fprintf(stderr, "nlabels must be equal to 2, got %d\n", nlabels);
+        rules_free(rules, nrules, 1);
+        rules_free(labels, nlabels, 0);
+        return 1;
+    }
+
+    if(nsamples_chk != nsamples) {
+        fprintf(stderr, "nsamples mismatch between out file (%d) and label file (%d)\n", nsamples, nsamples_chk);
+        rules_free(rules, nrules, 1);
+        rules_free(labels, nlabels, 0);
+        return 1;
+    }
 
     int nmeta, nsamples_check;
```

Comment:
Can we rename nsamples_check to something relating to meta

Reply:
Done (for nsamples_chk too)